sources/packages/web/src/views/aest/institution/Programs.vue

```diff
@@ -1,13 +1,216 @@
   </content-group>
 </template>
 
 <script lang="ts">
+import { onMounted, ref, computed } from "vue";
+import { useRouter } from "vue-router";
+import { InstitutionService } from "@/services/InstitutionService";
 import ContentGroup from "@/components/generic/ContentGroup.vue";
+import {
+  AESTInstitutionProgramsSummaryPaginatedDto,
+  ApprovalStatus,
+  SortDBOrder,
+} from "@/types";
+import { useToastMessage } from "@/composables";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
 export default {
   components: { ContentGroup },
-  //TODO, create and fetch an API to fetch Programs
+  props: {
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const toast = useToastMessage();
+    const router = useRouter();
+    const institutionProgramsSummary = ref(
+      {} as AESTInstitutionProgramsSummaryPaginatedDto,
+    );
+    const searchProgramName = ref("");
+    const DEFAULT_PAGE = 0;
+    const DEFAULT_ROW_SIZE = 10;
+    const DEFAULT_SORT_COLUMN = "submittedDate";
+    const DEFAULT_SORT_ORDER = SortDBOrder.DESC;
+    const loading = ref(false);
+    const getProgramsSummaryList = async (
+      institutionId: number,
+      rowsPerPage: number,
+      page: number,
+      sortColumn: string,
+      sortOrder: SortDBOrder,
+      programName: string,
+    ) => {
+      try {
+        loading.value = true;
+        searchProgramName.value = programName;
+        institutionProgramsSummary.value = await InstitutionService.shared.getPaginatedAESTInstitutionProgramsSummary(
+          institutionId,
+          rowsPerPage,
+          page,
+          sortColumn,
+          sortOrder,
+          programName,
+        );
+        if (institutionProgramsSummary.value.programsSummary.length === 0) {
+          toast.warn(
+            "No Programs found",
+            "No Programs found for the Institution/ Search Criteria",
+          );
+        }
+      } finally {
+        loading.value = false;
+      }
+    };
+    onMounted(async () => {
+      await getProgramsSummaryList(
+        props.institutionId,
+        DEFAULT_ROW_SIZE,
+        DEFAULT_PAGE,
+        DEFAULT_SORT_COLUMN,
+        DEFAULT_SORT_ORDER,
+        searchProgramName.value,
+      );
+    });
+    const programsFound = computed(() => {
+      return institutionProgramsSummary.value.programsCount > 0;
+    });
+    const goToViewProgramDetail = (programId: number) => {
+      router.push({
+        name: AESTRoutesConst.PROGRAM_DETAILS,
+        params: { programId: programId },
+      });
+    };
+    const getProgramStatusColorClass = (status: ApprovalStatus) => {
+      switch (status) {
+        case ApprovalStatus.approved:
+          return "bg-info text-white";
+        case ApprovalStatus.pending:
+          return "bg-warning text-white";
+        default:
+          return "";
+      }
+    };
+    const pageSortEvent = async (event: any) => {
+      getProgramsSummaryList(
```

Comment:
why await is not used? same for line 194.

Reply:
MIssed it, will add